scope.go

```diff
@@ -50,136 +50,117 @@ func NewScope() *Scope {
 // AddBreadcrumb adds new breadcrumb to the current scope
 // and optionaly throws the old one if limit is reached.
 func (scope *Scope) AddBreadcrumb(breadcrumb *Breadcrumb, limit int) {
+	scope.Lock()
+	defer scope.Unlock()
+
 	if breadcrumb.Timestamp == 0 {
 		breadcrumb.Timestamp = time.Now().Unix()
 	}
 
-	scope.Lock()
 	breadcrumbs := append(scope.breadcrumbs, breadcrumb)
 	if len(breadcrumbs) > limit {
 		scope.breadcrumbs = breadcrumbs[1 : limit+1]
 	} else {
 		scope.breadcrumbs = breadcrumbs
 	}
-	scope.Unlock()
 }
 
 // ClearBreadcrumbs clears all breadcrumbs from the current scope.
 func (scope *Scope) ClearBreadcrumbs() {
-	scope.Lock()
 	scope.breadcrumbs = []*Breadcrumb{}
-	scope.Unlock()
 }
 
 // SetUser sets new user for the current scope.
 func (scope *Scope) SetUser(user User) {
-	scope.Lock()
 	scope.user = user
-	scope.Unlock()
 }
 
 // SetRequest sets new user for the current scope.
 func (scope *Scope) SetRequest(request Request) {
-	scope.Lock()
 	scope.request = request
-	scope.Unlock()
 }
 
 // SetTag adds a tag to the current scope.
 func (scope *Scope) SetTag(key, value string) {
-	scope.Lock()
 	scope.tags[key] = value
-	scope.Unlock()
 }
 
 // SetTags assigns multiple tags to the current scope.
 func (scope *Scope) SetTags(tags map[string]string) {
 	scope.Lock()
+	defer scope.Unlock()
+
 	for k, v := range tags {
 		scope.tags[k] = v
 	}
-	scope.Unlock()
 }
 
 // RemoveTag removes a tag from the current scope.
 func (scope *Scope) RemoveTag(key string) {
-	scope.Lock()
 	delete(scope.tags, key)
-	scope.Unlock()
 }
 
 // SetContext adds a context to the current scope.
 func (scope *Scope) SetContext(key string, value interface{}) {
-	scope.Lock()
 	scope.contexts[key] = value
-	scope.Unlock()
 }
 
 // SetContexts assigns multiple contexts to the current scope.
 func (scope *Scope) SetContexts(contexts map[string]interface{}) {
 	scope.Lock()
+	defer scope.Unlock()
+
 	for k, v := range contexts {
 		scope.contexts[k] = v
 	}
-	scope.Unlock()
 }
 
 // RemoveContext removes a context from the current scope.
 func (scope *Scope) RemoveContext(key string) {
-	scope.Lock()
 	delete(scope.contexts, key)
-	scope.Unlock()
 }
 
 // SetExtra adds an extra to the current scope.
 func (scope *Scope) SetExtra(key string, value interface{}) {
-	scope.Lock()
 	scope.extra[key] = value
```

Comment:
Modifying a map, my guess is that u need a write lock for this collection too. Unless it's a thread-safe collection